solutions/USACO jan21-juststalling.mdx

```diff
@@ -0,0 +1,57 @@
+---
+id: usaco-1085
+source: USACO Bronze 2021 January
+title: Just Stalling
+author: Ananth Kashyap
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_prob3_bronze_jan21.html)
+
+For this problem, we are going to use permutations. First, we can sort the cows in increasing order based on
+height. Then we can keep track of the number of stalls that each cow is able to fit in. Now, to get the total
+number of possibilities that each cow can use in a permutation, all we have to do is, for each cow, subtract
+off the number of other cows that will be using other stalls which this cow is capable of using. To do this, we
+can use the fact that only cows with heights greater than or equal to this cow are required to use one of the
+stalls that this cow is capable of using. Since we know that our cows are sorted in height increasing order,
+the $\texttt{i}$th cow in our list will have $\texttt{n-i}$ cows with heights greater than or equal to it. So to
+get our final answer, we can initialize a cumulative product which will contain our answer, and for every
+$\texttt{i}$ between 1 and $\texttt{n}$, we just have to multiply this cumulative product by the number of
+stalls that the $\texttt{i}$th cow could possibly use, minus $\texttt{(n-i)}$.
+
+## Implementation
+
+Time Complexity: $\mathcal{O}(N^2)$
+
+<LanguageSection>
+
+<PySection>
+
+```python
+// Take in the data as an integer (n) and two separate lists (the stalls and the cows) using Python standard input.
```

Comment:
arent python comments # ? was this tested :/

Reply:
Yeah my bad I tested it and it was fine but then I added comments after (i thought nothing could go wrong) but i completely made this mistake. My bad i'll fix it right now.